### PR TITLE
London_10_Saqib_Javed_JavaScript-Core-2-Coursework-Week4

### PR DESCRIPTION
**Pull request by saqibjvd**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name:
- Your City:
- Your Slack Name:

# Homework Details

- Module:
- Week:

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

### CODE REVIEW

File: 1-mandatory/1-alarmclock/alarmclock.js [hunk 1]
```diff
@@ -1,4 +1,35 @@
...
+
+  let Interval = setInterval(() => {
+    totalSeconds -= 1;
+    dispalyRemainingTime(totalSeconds);
+    if (totalSeconds === 0) {
+      playAlarm();
+      clearInterval(Interval);
+    }
+  }, 1000);
+}
```
Comment (alismith-dev) [inline 1-mandatory/1-alarmclock/alarmclock.js:32]: Have you considered adding the `pause` functionality as a stretch goal?

File: 1-mandatory/1-alarmclock/alarmclock.js [hunk 2]
```diff
@@ -1,4 +1,35 @@
...
+
+function setAlarm() {
+  let inputTime = document.getElementById("alarmSet");
+  let totalSeconds = inputTime.value;
+  dispalyRemainingTime(totalSeconds);
+
+  let Interval = setInterval(() => {
+    totalSeconds -= 1;
+    dispalyRemainingTime(totalSeconds);
```
Comment (alismith-dev) [inline 1-mandatory/1-alarmclock/alarmclock.js:26]: Nice work! 🏅

File: 1-mandatory/1-alarmclock/alarmclock.js [hunk 3]
```diff
@@ -1,4 +1,35 @@
...
+    timeRemaining.textContent = "Time Remaining  " + minutes + ":" + seconds;
+  }
+}
+
+function setAlarm() {
+  let inputTime = document.getElementById("alarmSet");
+  let totalSeconds = inputTime.value;
+  dispalyRemainingTime(totalSeconds);
+
+  let Interval = setInterval(() => {
```
Comment (alismith-dev) [inline 1-mandatory/1-alarmclock/alarmclock.js:24]: This is just a minor thing, but i'd recommend using [`camelCase`](https://syllabus.codeyourfuture.io/guides/code-style-guide#use-camelcase) for most variables.

File: 1-mandatory/2-quotegenerator/quotes.js [hunk 4]
```diff
@@ -490,3 +490,19 @@ const quotes = [
...
+
+let quoteTitle = document.getElementById("quote");
+let authorName = document.getElementById("author");
+let button = document.getElementById("btn");
+let autoButton = document.getElementById("auto");
+
+function getQuotes() {
+  let randomQuote = pickFromArray(quotes);
+  quoteTitle.innerText = randomQuote.quote;
+  authorName.innerText = randomQuote.author;
```
Comment (alismith-dev) [inline 1-mandatory/2-quotegenerator/quotes.js:502]: This is really nice and clear, well done!

File: 1-mandatory/2-quotegenerator/quotes.js [hunk 5]
```diff
@@ -490,3 +490,19 @@ const quotes = [
...
+}
+
+button.addEventListener("click", getQuotes);
+getQuotes();
```
Comment (alismith-dev) [inline 1-mandatory/2-quotegenerator/quotes.js:506]: This works nicely, nice job 👏🏻

File: 1-mandatory/1-alarmclock/alarmclock.js [hunk 6]
```diff
@@ -1,4 +1,35 @@
...
+
+  // timeRemaining.textContent = "Time Remaining  " + minutes + ":" + seconds;
+
+  if (minutes < 10) {
+    timeRemaining.textContent =
+      "Time Remaining  " + "0" + minutes + ":" + seconds;
+  } else if (seconds < 10) {
+    timeRemaining.textContent =
+      "Time Remaining  " + minutes + ":" + "0" + seconds;
+  } else {
```
Comment (alismith-dev) [inline 1-mandatory/1-alarmclock/alarmclock.js:14]: There's a small issue with this logic: if you set the time to be 3 seconds, then it does not correctly format the time

Screenshot 2023-05-13 at 14 41 08

Can you figure out why this isn't working?

File: 1-mandatory/3-slideshow/slideshow.js [hunk 7]
```diff
@@ -1 +1,94 @@
...
+
+*/
+
+// variables
+
+const images = [
+  {
+    src: "./assets/cute-cat-a.png",
+    // alt : "cute white cat with black and grey stripes"
+    caption: "cute cat",
```
Comment (alismith-dev) [inline 1-mandatory/3-slideshow/slideshow.js:16]: This is a really nice bonus to add some `alt` text here to improve the accessibility of the slideshow 🙂 Can you figure out how to add this to the DOM?

File: 1-mandatory/3-slideshow/slideshow.js [hunk 8]
```diff
@@ -1 +1,94 @@
...
+];
+
+const image = document.querySelector("#carousel-img");
+const autoForwardBtn = document.querySelector("#auto-foward");
+const forwardBtn = document.querySelector("#forward-btn");
+const stopBtn = document.querySelector("#stop");
+const backwardBtn = document.querySelector("#backward-btn");
+const autoBackBtn = document.querySelector("#auto-backward");
+
+let ids = 0;
```
Comment (alismith-dev) [inline 1-mandatory/3-slideshow/slideshow.js:35]: Do you think the name of this variable really describes what it "means" in the code? Does it represent multiple ids? Or just one?

File: 1-mandatory/3-slideshow/slideshow.js [hunk 9]
```diff
@@ -1 +1,94 @@
...
+
+function getImageOnload() {
+  image.src = images[0].src;
+}
+
+function getNextPic() {
+  ids++;
+  if (ids > images.length - 1) {
+    ids = 0;
+  }
```
Comment (alismith-dev) [inline 1-mandatory/3-slideshow/slideshow.js:62]: This is great, nice work 🎆

File: 1-mandatory/3-slideshow/slideshow.js [hunk 10]
```diff
@@ -1 +1,94 @@
...
+  ids--;
+  if (ids < 0) {
+    ids = images.length - 1;
+  }
+  image.src = images[ids].src;
+}
+
+function showAutoForward() {
+  autoForwardBtn.disabled = true;
+  autoBackBtn.disabled = true;
```
Comment (alismith-dev) [inline 1-mandatory/3-slideshow/slideshow.js:76]: This is a really nice touch! 🙂

Comment (alismith-dev) [reply 1-mandatory/3-slideshow/slideshow.js:76]: One small thing I noticed is that if I click the Back/Forward buttons while the slideshow is auto moving forward/ back then it still moves the image.

What do you think we should do in that scenario? (It's a bit unclear exactly what the exercise expects you to do, but it's pretty normal a real job to not know all the details)

**Review by 40thieves — COMMENTED** [review]
Nice work @saqibjvd! 🍰🎉🕺🏻

**Pull request closed without merging**